### PR TITLE
New method for getting all comments from media

### LINKED ISSUE #591: All comments from media
### Purpose of your issue?
- [ ] Bug report (encountered problems/errors)
- [x] Feature request (request for a new functionality)
- [x] Question
- [ ] Other

---

### Get all comments from media

---

Code:

```python
def get_media_comments(bot, media_id):
    result = []
    for i in range(10):
        q1 = '?query_hash=f0986789a5c5d17c2400faebf16efd0d&variables={"shortcode"%3A"Bk0kC3GFWgq"%2C"first"%3A'
        q2 = '%2C"after"%3A"AQB7DjR2XIDcGM-xC6AMy7EvhO7lsXn1hgUFTME2Vu-zrxmb7ipiC22ADOxUC242kGdKBWcTOcR3J9-0_LfoYWeVXpnqsmhxFmAvSMrBG2VeyQ"}'
        url = 'media/{media_id}/comments/'.format(media_id=media_id)
        # url += q1 + str(i*100) + q2 # I try this but it was same as before
        url += '?page%s' % i
        bot.api.send_request(url)
        last_item = bot.api.last_json
        result += last_item
```

---

### Describe your issue

I want to get all comments from media, I saw graph query language from an online website but I couldn't do the same in api system
Do you know how I should do this?
my post have more than 10000 comments

### PR DESCRIPTION
**Pull request by ademaro**
...or what you want by param `count`

this PR fix #591

### CODE REVIEW

File: instabot/bot/bot_get.py [hunk 1]
```diff
@@ -215,6 +216,30 @@ def get_media_comments(self, media_id, only_text=False):
...
+        for comment in self.api.last_json['comments']:
+            comments.append(comment)
+        has_more_comments = self.api.last_json['has_more_comments']
+        if count and len(comments) >= count:
+            comments = comments[:count]
+            has_more_comments = False
+            self.logger.info("Getting comments stopped by count (%s)." % count)
+        if has_more_comments:
+            max_id = self.api.last_json['next_max_id']
+            time.sleep(2)
```
Comment (ademaro) [inline instabot/bot/bot_get.py]: I'm not sure about this...

Comment (ohld) [reply instabot/bot/bot_get.py]: Have you tested this to be sure?

**Comment by ademaro** [comment]
@ohld Why not accept PR? Do I need to implement new unit tests for this case?

File: instabot/bot/bot_get.py (hunk 1 above)
Comment (basnijholt) [reply instabot/bot/bot_get.py]: This sleep is not needed IIUC

**Comment by ohld** [comment]
@basnijholt @sudoguy I'm going to merge. Any thoughts?

**Review by ohld — APPROVED** [review]
Nice Work!

**Comment by ohld** [comment]
Sorry for the delay! Great work!

**Pull request merged**